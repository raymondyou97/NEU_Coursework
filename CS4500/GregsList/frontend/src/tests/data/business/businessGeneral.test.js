'use es6';

import React from 'react';
import BusinessGeneral from '../../../components/business/General';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

test('Initial snapshot renders', () => {
  const component = renderer.create(<BusinessGeneral />);

  let componentTree = component.toJSON();
  expect(componentTree).toMatchSnapshot();
});

test('Correct number of default forms and default values', () => {
  const component = shallow(<BusinessGeneral />);

  expect(component.find('.form-control').length).toEqual(4);
  expect(component.find('#businessNameInput').props().value).toEqual(undefined);
  expect(component.find('#businessYearFoundedInput').props().value).toEqual(
    undefined
  );
  expect(component.find('#businessNumEmployeesInput').props().value).toEqual(
    undefined
  );
  expect(component.find('#businessEmailInput').props().value).toEqual(
    undefined
  );
});
