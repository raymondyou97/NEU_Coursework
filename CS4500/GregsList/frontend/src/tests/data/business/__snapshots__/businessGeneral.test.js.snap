// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Initial snapshot renders 1`] = `
<div>
  <h5>
    Business General
  </h5>
  <div
    className="row"
  >
    <div
      className="col-12"
    >
      <label
        htmlFor="businessNameInput"
      >
        Business name
      </label>
      <input
        className="form-control"
        id="businessNameInput"
        type="text"
      />
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col-12"
    >
      <label
        htmlFor="businessYearFoundedInput"
      >
        Year founded
      </label>
      <input
        className="form-control"
        id="businessYearFoundedInput"
        type="number"
      />
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col-12"
    >
      <label
        htmlFor="businessNumEmployeesInput"
      >
        Number of employees
      </label>
      <input
        className="form-control"
        id="businessNumEmployeesInput"
        type="number"
      />
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col-12"
    >
      <label
        htmlFor="businessEmailInput"
      >
        Email
      </label>
      <input
        className="form-control"
        id="businessEmailInput"
        type="email"
      />
    </div>
  </div>
</div>
`;
